crates/cast/bin/tx.rs

```diff
@@ -182,6 +183,20 @@ where
             tx.set_nonce(nonce.to());
         }
 
+        if let Some(access_list) = match tx_opts.access_list {
```

Comment:
we should do it right before estimating gas in `_build`

we don't yet have calldata here, so calling rpc will result in incomplete data or an error